app/controlplane/extensions/core/smtp/v1/extension.go

```diff
@@ -0,0 +1,236 @@
+		return nil, fmt.Errorf("invalid attachment request: %w", err)
+	}
+
+	var rc *registrationState
+	if err := sdk.FromConfig(req.RegistrationInfo.Configuration, &rc); err != nil {
+		return nil, errors.New("invalid registration configuration")
+	}
+
+	response := &sdk.AttachmentResponse{}
+	rawConfig, err := sdk.ToConfig(&attachmentState{Cc: request.Cc})
+	if err != nil {
+		return nil, fmt.Errorf("marshalling configuration: %w", err)
+	}
+	response.Configuration = rawConfig
+
+	return response, nil
+}
+
+// Send the SBOM to the configured Dependency Track instance
+func (i *Integration) Execute(_ context.Context, req *sdk.ExecutionRequest) error {
+	i.Logger.Info("execution requested")
+
+	if err := validateExecuteRequest(req); err != nil {
+		return fmt.Errorf("running validation for workflow id %s: %w", req.WorkflowID, err)
+	}
+
+	var rc *registrationState
+	if err := sdk.FromConfig(req.RegistrationInfo.Configuration, &rc); err != nil {
+		return errors.New("invalid registration configuration")
+	}
+
+	var ac *attachmentState
+	if err := sdk.FromConfig(req.AttachmentInfo.Configuration, &ac); err != nil {
+		return errors.New("invalid attachment configuration")
+	}
+
+	// get the attestation
+	decodedPayload, err := req.Input.DSSEnvelope.DecodeB64Payload()
+	if err != nil {
+		return err
+	}
+	statement := &in_toto.Statement{}
+	if err := json.Unmarshal(decodedPayload, statement); err != nil {
+		return fmt.Errorf("un-marshaling predicate: %w", err)
+	}
+	jsonBytes, err := json.MarshalIndent(statement, "", "  ")
+	i.Logger.Info("statement", string(jsonBytes))
+	if err != nil {
+		fmt.Println("Error marshaling JSON:", err)
+		return err
+	}
+
+	// send the email
+	to, from, user, password, host, port := rc.To, rc.From, rc.User, req.RegistrationInfo.Credentials.Password, rc.Host, rc.Port
+	subject := "[chainloop] New workflow run finished successfully!"
+	tpl := `A new workflow run finished successfully!
+
+# Workflow: %s
+
+# in-toto statement:
+	%s
+
+This email has been delivered via integration %s version %s.
+	`
+	body := fmt.Sprintf(tpl, req.WorkflowID, jsonBytes, i.Describe().ID, i.Describe().Version)
+	err = sendEmail(host, port, user, password, from, to, ac.Cc, subject, body)
+	if err != nil {
+		return fmt.Errorf("sending an email: %w", err)
+	}
+
+	return nil
+}
+
+func validateExecuteRequest(req *sdk.ExecutionRequest) error {
+	if req == nil || req.Input == nil || req.Input.DSSEnvelope == nil {
+		return errors.New("invalid input")
+	}
+
+	if req.RegistrationInfo == nil || req.RegistrationInfo.Configuration == nil {
+		return errors.New("missing registration configuration")
+	}
+
+	if req.RegistrationInfo.Credentials == nil {
+		return errors.New("missing credentials")
+	}
+
+	if req.AttachmentInfo == nil || req.AttachmentInfo.Configuration == nil {
+		return errors.New("missing attachment configuration")
+	}
+
+	return nil
+}
+
+func sendEmail(host string, port string, user, password, from, to, cc, subject, body string) error {
```

Comment:
when you start having more than 2 arguments, specially if they are from the same type, it's common to change the function signature to accept a struct instead. That makes it more readable and less error prone. I'd suggest doing that here

Reply:
yes, this will change once we move it out to a separate package